### ISSUE TITLE
Registration Cache Invalidation?

### ISSUE THREAD

**Issue by Mike-EEE** [unitycontainer/unity#47]
Hello,

Consider the following component:

```
public class Component
{
    public Component( IDependency dependency ) { ... }

    public Dependency Dependency { get; }
}

public class Dependency : IDependency { ... }
public class AnotherDependency : IDependency { ... }
```

Now let's say that at application initialization, `Dependency` is registered as the `IDependency` and the `Component` is created:

```
var container = new UnityContainer();
container.RegisterType<IDependency, Dependency>();
container.Register<Component>( new ContainerControlledLifetimeManager() );
var instance = container.Resolve<Component>();
```

Now let's I do this:

```
container.RegisterType<IDependency, AnotherDependency>();
```

What I would like to happen is to essentially invalidate the lifetime managers for all registrations that have direct (or indirect) dependencies to `IDependency`.  So, if I do this:

```
var instance = container.Resolve<Component>();
Assert.IsType<AnotherDependency>( instance.Dependency ); // Equals true
```

I'm using psedo-code here, but hopefully you can get what I am driving at.  Is this possible?  Is there an extension out there that already does this?  Any insight/guidance would be appreciated.

**Comment by Michael-DST** [comment]
Sounds like a "no."  :P  Is this repo alive/dead or what?

**Comment by ENikS** [comment]
This would take some research.

**Comment by ENikS** [comment]
This is fixed in 5.0.0
